skimage/filters/thresholding.py

```diff
@@ -335,15 +337,20 @@ def threshold_li(image):
     >>> thresh = threshold_li(image)
     >>> binary = image > thresh
     """
+    assert_nD(image, 2)
     # Requires positive image (because of log(mean))
-    offset = image.min()
-    # Can not use fixed tolerance for float image
-    imrange = image.max() - offset
-    image -= offset
+    offset = img.min()
+    if offset < 0:
```

Comment:
@JDWarner the image doesn't need to be offset if it's non-negative (see the `else` clause). @sciunto but the range needs to be updated to subtract the minimum. I also would rather we keep using `image` instead of `img`. (Then you don't need the `else`.)